neqo-transport/src/frame.rs

```diff
@@ -426,8 +426,13 @@ impl<'a> Frame<'a> {
             d(dec.decode_varint())
         }
 
-        // TODO(ekr@rtfm.com): check for minimal encoding
+        // Check for minimal encoding of frame type.
+        let pos = dec.offset();
         let t = dv(dec)?;
+        if Decoder::minimal_varint_len(t).unwrap() != dec.offset() - pos {
```

Comment:
```suggestion
        // RFC 9000, Section 12.4:
        //
        // The Frame Type field uses a variable-length integer encoding [...],
        // with one exception. To ensure simple and efficient implementations of
        // frame parsing, a frame type MUST use the shortest possible encoding.
        if Decoder::minimal_varint_len(t).unwrap() != dec.offset() - pos {
```

Maybe worth documenting why a minimal encoding is a MUST. Feel free to ignore.
```